src/test/scala/com/amazon/deequ/analyzers/runners/AnalysisRunnerTests.scala

```diff
@@ -271,5 +271,37 @@ class AnalysisRunnerTests extends WordSpec with Matchers with SparkContextSpec w
         inputStream => assert(inputStream.read() > 0)
       }
     }
+
+    "should save states when running on aggregated states" in withSparkSession { session =>
+
+      val dataFrames = getDFsWithNRowsAndUniqueValuesCrossDF(session, 2, 2)
+
+      val stateStores = Seq(InMemoryStateProvider(), InMemoryStateProvider())
+
+      val analysis = Analysis().addAnalyzers(Size() :: Uniqueness("c1") :: Nil)
+
+      for((dataFrame, stateStore) <- dataFrames zip stateStores) {
```

Comment:
I dont quite understand this code. What are the semantics of ```dataFrames``` here? Is it different versions of the same data? Or is a growing dataset such that the union of all the data frames makes up the dataset?

Reply:
this is two different dataframes with two different set of data,  Then a different stateStore is generated and linked for each dataframe.

After,  metrics for each independent dataFrame are calculated and the state stored on its linked stateStore.
(It could be seen as Monday data and metrics for Monday on same for Tuesday).

Then this states are aggregated with AnalysisRunner.runOnAggregatedStates without any new data, just the previous saved states and saved on a new StateProvider. (Which would represent the Monday + Tuesday aggregated and stored).